autocti/extract/two_d/abstract.py

```diff
@@ -215,29 +270,35 @@ def binned_array_1d_from(
         )
 
     def binned_array_1d_total_pixels_from(
-        self, array: aa.Array2D, pixels: Tuple[int, int]
+        self,
+        array: aa.Array2D,
+        pixels: Optional[Tuple[int, int]] = None,
+        pixels_from_end: Optional[int] = None,
     ) -> aa.Array1D:
         """
-            The function `binned_array_1d_from` extracts a region (e.g. the parallel FPR) of every charge injection region
-            on the charge injection image, stacks them by taking their mean and then bin them up to a 1D region
-            (e.g. the 1D parallel FPR) by taking the mean across the direction opposite to clocking (e.g. bin over the
-            serial direction for a parallel FPR).
+        The function `binned_array_1d_from` extracts a region (e.g. the parallel FPR) of every charge injection region
+        on the charge injection image, stacks them by taking their mean and then bin them up to a 1D region
+        (e.g. the 1D parallel FPR) by taking the mean across the direction opposite to clocking (e.g. bin over the
+        serial direction for a parallel FPR).
 
-            If the data being stacked is a noise-map, we need to know how many pixels were used in the stacking of every
-            final pixel on the binned 1D array in order to compute the new noise map via quadrature.
+        If the data being stacked is a noise-map, we need to know how many pixels were used in the stacking of every
+        final pixel on the binned 1D array in order to compute the new noise map via quadrature.
 
-            Parameters
+        Parameters
         ----------
-            array
-                The 2D array which contains the charge injeciton image from which the parallel FPRs are extracted and
-                stacked.
-            pixels
-                The column / row pixel index to extract the region (e.g. FPR, EPER) between (e.g. `pixels=(0, 3)` extracts
-                the 1st, 2nd and 3rd columns / rows)
+        array
+            The 2D array which contains the charge injeciton image from which the parallel FPRs are extracted and
+            stacked.
+        pixels
+            The column / row pixel index to extract the region (e.g. FPR, EPER) between (e.g. `pixels=(0, 3)` extracts
+            the 1st, 2nd and 3rd columns / rows)
         """
 
         mask_2d_list = [
-            array.mask[region.slice] for region in self.region_list_from(pixels=pixels)
+            array.mask[region.slice]
+            for region in self.region_list_from(
+                pixels=pixels, pixels_from_end=pixels_from_end

```

Comment:
That pixels from end to pixels decorator would save so many changes. In particular I would suggest using it at whatever point you first call a function. 

Even if you weren't using the decorator doing the conversion then passing pixels rather than two optional arguments to all of these functions would be much cleaner. You've had to make an absolute ton of changes just to support having one of two possible arguments between which there is a simple relationship.